include/Dialects/LinalgTransform/TransformOpInterface.h

```diff
@@ -64,6 +105,10 @@ class TransformState {
 
   /// The mapping between payload IR values and transform IR ops.
   TransformOpMapping operations;
+
+  /// Extensions attached to the TransformState, identified by the TypeID of
+  /// their type. Only one extension of any given type is allowed.
+  DenseMap<TypeID, std::unique_ptr<Extension>> extensions;
```

Comment:
I do not understand the need for this complexity, AFAIU `TypeID` is always `TypeID::get<TrackingState>`.

Is this premature generalization?

Reply:
The same as for having all the `functional::` stuff: separation of concerns and preparation for upstreaming.